### PR TITLE
Add JSON support for formatting output data

### PR DESCRIPTION
**Pull request by btaczala**
This commit adds JSON support for formatting results. I'd like to hook fccf into other tools (neovim) and it's much easier to parse JSON that plain text. 

This adds a 3rdParty dependency (awesome [nlohmann json](https://github.com/nlohmann/json) library) - hope you won't mind.

### CODE REVIEW

**Comment by p-ranav** [comment]
Hello @btaczala. Thanks for the PR. I like the idea and I want to merge it. I've made some minor suggestions (style related) and asked a few questions. Once these are resolved, we can merge. Thanks.

**Review by p-ranav — CHANGES_REQUESTED** [review]

File: source/searcher.cpp [hunk 1]
```diff
@@ -7,12 +7,43 @@ namespace fs = std::filesystem;
 
 #include <clang-c/Index.h>  // This is libclang.
 
+namespace
+{
+void printPlainText(std::string_view filename,
```
[the remark this answers is not in the document]
Comment (btaczala) [reply source/searcher.cpp]: Fixed in fdf3c7091bd8f26af86270107a27029b7a4e3561

File: source/main.cpp [hunk 2]
```diff
@@ -345,6 +354,22 @@ int main(int argc, char* argv[])
   searcher.m_search_for_for_statement = no_filter || search_for_for_statement;
   searcher.m_ignore_single_line_results = ignore_single_line_results;
   searcher.m_ts = std::make_unique<thread_pool>(num_threads);
+  nlohmann::json jsonArray = nlohmann::json::array();
```
[the remark this answers is not in the document]
Comment (btaczala) [reply source/main.cpp]: Fixed in 17778c2613ffd5e42bb01472f0c9bacdaf9644a9

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -4,10 +4,12 @@
 
 project(fccfTests LANGUAGES CXX)
 
+find_package(Clang REQUIRED CONFIG)
+
 # ---- Tests ----
 
 add_executable(fccf_test source/fccf_test.cpp)
-target_link_libraries(fccf_test PRIVATE fccf_lib)
+target_link_libraries(fccf_test PRIVATE fccf_lib libclang)
```
[the remark this answers is not in the document]
Comment (btaczala) [reply test/CMakeLists.txt]: Fixed in 17778c2613ffd5e42bb01472f0c9bacdaf9644a9

File: test/CMakeLists.txt [hunk 4]
```diff
@@ -4,10 +4,12 @@
 
 project(fccfTests LANGUAGES CXX)
 
+find_package(Clang REQUIRED CONFIG)
```
[the remark this answers is not in the document]
Comment (btaczala) [reply test/CMakeLists.txt]: Fixed in 17778c2613ffd5e42bb01472f0c9bacdaf9644a9

File: source/main.cpp [hunk 5]
```diff
@@ -9,6 +9,9 @@
 #include <argparse.hpp>
 #include <searcher.hpp>
 #include <unistd.h>
+
+#include "json.hpp"
```
[the remark this answers is not in the document]
Comment (btaczala) [reply source/main.cpp]: Fixed in fdf3c7091bd8f26af86270107a27029b7a4e3561

**Review by p-ranav — APPROVED** [review]

**Pull request merged**